src/components/windows-motion-controls.js

```diff
@@ -0,0 +1,382 @@
+          pressed: getImmediateChildByName(buttonMesh, 'PRESSED'),
+          unpressed: getImmediateChildByName(buttonMesh, 'UNPRESSED')
+        };
+        if (buttonMeshInfo.value && buttonMeshInfo.pressed && buttonMeshInfo.unpressed) {
+          loadedMeshInfo.buttonMeshes[this._mapping.buttons[i]] = buttonMeshInfo;
+        } else {
+          // If we didn't find the mesh, it simply means this button won't have transforms applied as mapped button value changes.
+          warn('Missing button submesh under mesh with name: ' + buttonMeshName +
+            '(VALUE: ' + !!buttonMeshInfo.value +
+            ', PRESSED: ' + !!buttonMeshInfo.pressed +
+            ', UNPRESSED:' + !!buttonMeshInfo.unpressed +
+            ')');
+        }
+      }
+
+      // Axis Meshes
+      for (i = 0; i < this._mapping.axisMeshNames.length; i++) {
+        var axisMeshName = this._mapping.axisMeshNames[i];
+        if (!axisMeshName) {
+          debug('Skipping unknown axis at index: ' + i);
+          continue;
+        }
+
+        var axisMesh = rootNode.getObjectByName(axisMeshName);
+        if (!axisMesh) {
+          warn('Missing axis mesh with name: ' + axisMeshName);
+          continue;
+        }
+
+        var axisMeshInfo = {
+          index: i,
+          value: getImmediateChildByName(axisMesh, 'VALUE'),
+          min: getImmediateChildByName(axisMesh, 'MIN'),
+          max: getImmediateChildByName(axisMesh, 'MAX')
+        };
+        if (axisMeshInfo.value && axisMeshInfo.min && axisMeshInfo.max) {
+          loadedMeshInfo.axisMeshes[i] = axisMeshInfo;
+        } else {
+          // If we didn't find the mesh, it simply means this axis won't have transforms applied as mapped axis values change.
+          warn('Missing axis submesh under mesh with name: ' + axisMeshName +
+            '(VALUE: ' + !!axisMeshInfo.value +
+            ', MIN: ' + !!axisMeshInfo.min +
+            ', MAX:' + !!axisMeshInfo.max +
+            ')');
+        }
+      }
+    } else {
+      warn('No node with name "RootNode" in controller glTF.');
+    }
+
+    debug('Model load complete.');
+
+    // Look through only immediate children. This will return null if no mesh exists with the given name.
+    function getImmediateChildByName (object3d, value) {
+      for (var i = 0, l = object3d.children.length; i < l; i++) {
+        var obj = object3d.children[i];
+        if (obj && obj['name'] === value) {
+          return obj;
+        }
+      }
+      return undefined;
+    }
+  },
+
+  lerpAxisTransform: function (axis, axisValue) {
+    var axisMeshInfo = this._loadedMeshInfo.axisMeshes[axis];
+    if (!axisMeshInfo) return;
+
+    var min = axisMeshInfo.min;
+    var max = axisMeshInfo.max;
+    var target = axisMeshInfo.value;
+
+    // Convert from gamepad value range (-1 to +1) to lerp range (0 to 1)
+    var lerpValue = axisValue * 0.5 + 0.5;
+    target.setRotationFromQuaternion(min.quaternion.clone().slerp(max.quaternion, lerpValue));
```

Comment:
To appease the garbage collector we should avoid allocating new variables and reuse instead as much as we can. We usually use a closure pattern for temporary variables that don't hold state (https://github.com/aframevr/aframe/blob/master/src/components/cursor.js#L168). In this case I think we could do:
````javascript
lerpAxisTransform: (function () {
 var quaternion = new THREE.Quaternion();
 return function (axis, axisValue) {
   ....
      target.setRotationFromQuaternion(quaternion.copy(min.quaternion).slerp(max.quaternion, lerpValue));
   ....
 }
});

Reply:
my personal preference is to have them as unpublicized properties on the component instance, so that other functions can reuse them, but yes I think the vast majority of code uses closure form

Reply:
Two things:
1. If it's just a temporary value is better to use a closure. By having the variable on the closure you share one single variable across all the object instances.
2. In the `this`. it pollutes the object and it's tempting to use it to store state that will be destroyed if multiple methods use the variable for partial calculations.
3. If those variables are not longer useful they will be hanging from the `this`. it's easy to forget and leave them there orphan.

Reply:
Completely agree with 2 and 3.
On 1, I thought we ran into past issues with multiple instances / invocations there, but may be misremembering.